### PR TITLE
add rgb led configuration for xd87

### PR DESCRIPTION
**Pull request by LouWii**
## Description

Added configuration for RGB underglow for the XD87. Disable debug features to save space.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage). (compiled the firmware for the keyboard and tested it)

### CODE REVIEW

**Comment by drashna** [comment]
Is there a specific reason to enable the rgb globally, rather than just for a specific keymap?

**Comment by LouWii** [comment]
I thought underglow would be used by most people, therefore adding it for everybody. I don't mind adding it to a specific keymap, but not sure what would be the best practice in that case. Should I call the keymap `rgb-underglow` and add a second layer with RGB underglow functions keys? (toggle, hue+/-, brightness+/- ans so on)

**Comment by fauxpark** [comment]
It seems like this board does not come with the RGB LEDs populated.

It might be best to leave it disabled in the default rules.mk, to save flash space, and either let users turn it on in their own keymaps if they've soldered the LEDs on, or create a `default_rgb` keymap with it enabled in its rules.mk.

**Comment by yanfali** [comment]
Underglow is probably not that useful to most people, given the default case options, but there's space in the flash so it does no harm and makes configuring rgb from the configurator easier.

**Comment by LouWii** [comment]
I had to disable either `console` or `command` to make it all fit with a 2 layers layout. I've disabled both `console` and `command` in my commit as I'm not sure it's used by most people (feel free to prove me wrong :) ).

I agree it might also be the case of the RGB underglow as they aren't populated by default and there's no case taking advantage of it. I will create a `default_rgb` layout tomorrow with rgb enabled, and update the default `rules.mk` to disable it by default. @yanfali people using the configurator will have to start from the `default_rgb` layout which should be ok.

**Comment by yanfali** [comment]
> I had to disable either `console` or `command` to make it all fit with a 2 layers layout. I've disabled both `console` and `command` in my commit as I'm not sure it's used by most people (feel free to prove me wrong :) ).
> 
> I agree it might also be the case of the RGB underglow as they aren't populated by default and there's no case taking advantage of it. I will create a `default_rgb` layout tomorrow with rgb enabled, and update the default `rules.mk` to disable it by default. @yanfali people using the configurator will have to start from the `default_rgb` layout which should be ok.

If you had to disable console and command, I'd rather you actually not enable RGB then for the default. Those are very useful for debugging.

**Comment by yanfali** [comment]
@LouWii I'd rather you disable bootmagic actually and just set it to `lite`

**Comment by LouWii** [comment]
> @LouWii I'd rather you disable bootmagic actually and just set it to `lite`

I don't mind doing that.

I'm still not sure about enabling both command and console. Do people usually use them? If we're saying RGB should be disabled by default because most people won't use it, it seems that console and command are almost in the same position. I'm pretty new in the QMK world though, I can be wrong. I looked at a few keyboards `rules.mk` and it's quite a mix out there, it's either one, the other, both or none.

**Comment by yanfali** [comment]
@LouWii we leave it to users to override their settings if they need to, but for default, it's very nice to have command and console enabled because you can do things like get the version of QMK and reset the eeprom, which is really good for debugging purposes when we're helping people. Since the focus of the xd87 isn't rgb and it doesn't come with by default, it's better for us as collaborators to leave those things off in favor of things which can be useful.

**Comment by LouWii** [comment]
Alright, I've made the changes as we discussed. RGB has been removed from the default config, command and console are back. BTW, the documentation recommends to disable bootmagic and command by default: https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=bootmagic-and-command

I've added a keymap with RGB enabled, command and bootmagic disabled (I could get away with lite bootmagic but it was leaving very few free space, same with disabling bootmagic entirely, so that's why I decided to disable command as well).

Compiled and tested on my own board, works fine

File: keyboards/xd87/keymaps/rgb_default/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 Louwii
```
Comment (fauxpark) [inline keyboards/xd87/keymaps/rgb_default/config.h]: This keymap should be named `default_rgb` or `default_underglow`, actually. I'm leaning towards the latter as there are two RGB-related features (Underglow and Matrix) and that would help to disambiguate.

File: keyboards/xd87/keymaps/rgb_default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  QMKBEST = SAFE_RANGE,
+  QMKURL
+};
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_tkl_ansi(\
```
Comment (fauxpark) [inline keyboards/xd87/keymaps/rgb_default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/xd87/keymaps/rgb_default/rules.mk [hunk 3]
```diff
@@ -0,0 +1,3 @@
+BOOTMAGIC_ENABLE = no
```
Comment (fauxpark) [inline keyboards/xd87/keymaps/rgb_default/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = lite
```

File: keyboards/xd87/keymaps/rgb_default/readme.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# The default RGB underglow layout for xd87
+
+The RGB underglow feature is disabled by default. This specific keymap enables RGB underglow but disables _command_ and _bootmagic_ in order to save space.
```
Comment (fauxpark) [inline keyboards/xd87/keymaps/rgb_default/readme.md]:
```suggestion
The RGB Underglow feature is disabled by default. This specific keymap enables RGB Underglow but disables the Command feature in order to save space.
```

File: keyboards/xd87/keymaps/rgb_default/config.h (hunk 1 above)
Comment (LouWii) [reply keyboards/xd87/keymaps/rgb_default/config.h]: Renamed it `default_underglow`

File: keyboards/xd87/keymaps/rgb_default/keymap.c (hunk 2 above)
Comment (LouWii) [reply keyboards/xd87/keymaps/rgb_default/keymap.c]: Interesting. I've seen them pretty much everywhere, I thought C wouldn't like the carriage return. I've removed them.

File: keyboards/xd87/keymaps/rgb_default/rules.mk (hunk 3 above)
Comment (LouWii) [reply keyboards/xd87/keymaps/rgb_default/rules.mk]: Done

File: keyboards/xd87/keymaps/rgb_default/readme.md (hunk 4 above)
Comment (LouWii) [reply keyboards/xd87/keymaps/rgb_default/readme.md]: Done.

File: keyboards/xd87/keymaps/rgb_default/keymap.c (hunk 2 above)
Comment (fauxpark) [reply keyboards/xd87/keymaps/rgb_default/keymap.c]: They're only needed for the actual layout `#define`, as the preprocessor is expecting it to be all on one line, and the backslashes tell it to ignore the newline and keep going.

**Pull request merged**